src/main/scala/firrtl/Emitter.scala

```diff
@@ -330,6 +331,7 @@ class VerilogEmitter extends SeqTransform with Emitter {
        case DoPrim(op, args, _,_) if isCast(op) => args.foreach(checkArgumentLegality)
        case DoPrim(op, args, _,_) if isBitExtract(op) => args.foreach(checkArgumentLegality)
        case DoPrim(Cat, args, _, _) => args foreach(checkCatArgumentLegality)
+       case DoPrim(op, args, _,_) if isBooleanExpr(op) => args.foreach(checkArgumentLegality)
```

Comment:
This will emit illegal code if context-dependent widths exist in the emitted Verilog expression.

Reply:
Isn't it worse than illegal code? It is legal code that implements different functionality?

Reply:
Yeah, my bad, by "illegal" I mean "wrong."